### DISCUSSION TITLE
Problem accessing SD Card on PyBoard v1.1 clone

### DISCUSSION THREAD

**Discussion by MarkMakies** [micropython/micropython#15093] [STM32 / Pyboard] [answered]
Hi,

After much effort and countless searches I have been unable to mount an SD card on my PyBoard v1.1 clone.  I've tried 6 different cards, formatted in many ways (using both disks and gparted), 1 or 2 partitions, FAT 16 & 32, / msdos, GPT &  loop partition tables, different sizes 2/8/32/64 and different brands Transcend, Kingston , SanDisk and Pico.

The Pyboard is listed on eBay as 'unbranded'.  It appears to work well in every other way.

I've also tried formatting from MPY to no avail in both FAT and Little FS v2. (Using  vfs.VfsLfs2.mkfs)

My only lead at this point is that  executing sd.power(True) returns False and sd.power(False) returns True.  I have no idea what this means and haven't been able to find more info?

Any help would be appreciated.  Thanks,
Mark

```
MPY: can't mount SD card
MicroPython v1.22.2 on 2024-02-22; PYBv1.1 with STM32F405RG
Type "help()" for more information.

>>> import os, pyb
>>> sd = pyb.SDCard()
>>> os.mount(sd, '/sd')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: 16

>>> sd.power(True)
False

>>> pyb.SD.power(False)
True

>>> sd.present()
True

>>> os.VfsLfs2.mkfs(pyb.SDCard())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: 28
>>> 

```

**Comment by peterhinch** [comment]
The SD card should automatically be mounted. On a real pyboard after power up
```py
MicroPython v1.22.1 on 2024-01-05; PYBv1.1 with STM32F405RG
Type "help()" for more information.
>>> import os
>>> os.listdir("/")
['flash', 'sd']
>>> 
```
This is without any user code. `boot.py`:
```py
import pyb
pyb.usb_mode('VCP')
```
The SD card needs no special formatting. A standard, brand new, SD card should just work.

**Comment by MarkMakies** [comment]
Hi Peter,

Thanks for your response.  I suspect that this clone has a problem.  I've read where others have found the SD_PRESENT switch mapped to PA13 rather than PA8, or not connected at all.   pyb.sd.present() appears to work as it should for me.  My only clue now is that it is not getting powered up properly

```
>>> pyb.sd.power(True)
False
>>> pyb.SD.power(False)
True
```
To me this looks like this function triggers logic to provide power to the SD once detected as present, and I guess the return value indicates if power actually is enabled on the device.  I've read through the relevant part of the c source code, but it is beyond my understanding.  I can't find a 'power' pin in the pins.csv list, but it does show...

```
SD_D0	PC8
SD_D1	PC9
SD_D2	PC10
SD_D3	PC11
SD_CMD   PD2
SD_CK	   PC12
SD	           PA8
SD_SW	   PA8
```
PA8 is mapped top two pins - should 1 be for power?

Thanks,
mark

PS. Many thanks Peter for all your posts.  I've been learning micropython for two years now, and using it on multiple platforms - every time I have an issue, I can nearly always find the answer in one of your posts.

**Comment by MarkMakies** [comment]
I've checked the PyBoard v1.1 schematic and note that the SD card slot is always powered.  So what does pyb.SDCard.power() actually do?   Whilst trying to find a solution I found that when others executed pyb.SDCard.power(True) it returns True.  Mine always returns false.

![image](https://github.com/micropython/micropython/assets/105891859/ad6c5336-58a7-46be-8b4f-a602ce39f6ad)

**Reply by peterhinch** [reply]
It seems highly likely that there is a hardware fault in your board, either a design flaw or a random fail. You may find that the most cost effective solution is to buy a [genuine pyboard](https://store.micropython.org/product/PYBv1.1) and support MicroPython development.

**Reply by MarkMakies** [reply, marked as answer]
Thanks Peter, 

Definitely a design fault, I bought a handful and all exhibit the same problem.  I agree that I'm not pursuing the most cost effective solution, but I'm retired, have time and sometimes the thrill is in uncovering, understanding and solving difficult problems.   I'll buy a few genuine boards today which will also serve as a reference whilst supporting the excellent MicroPython development that I have gained so much from.

Anyway I got out the microscope and had a good look at the board.  It appears SD card slot, pin 8 (DATA1), is missing a solder pad.  It's awfully small but I'm going to try to connect to pin 40  on the MCU.  

![image](https://github.com/micropython/micropython/assets/105891859/d2f6d7c1-c8f9-48d2-bf8b-772c1b207df4)
